test/gtest/uct/ib/test_ib_pkey.cc

```diff
@@ -201,14 +211,26 @@ UCS_TEST_P(test_uct_ib_pkey, test_pkey_pairs) {
         /* pack-unpack the first IB iface address */
         uct_ib_iface_t *iface1      = ucs_derived_of(m_e1->iface(),
                                                      uct_ib_iface_t);
-         uct_ib_address_t *ib_addr1 =
+        status = uct_iface_query(m_e1->iface(), &iface_attr);
+        ASSERT_UCS_OK(status);
+        iface_addr1.resize(iface_attr.iface_addr_len);
+        status = uct_iface_get_address(m_e1->iface(),
+                                       (uct_iface_addr_t*)iface_addr1.data());
+        ASSERT_UCS_OK(status);
```

Comment:
maybe add wrapper function to get iface addr as vector

Reply:
reused existing `auto_buffer` class + minor improvements